src/common-opencl.c

```diff
@@ -1252,13 +1251,24 @@ static cl_ulong gws_test(size_t gws, unsigned int rounds, int sequential_id)
 
 	// Set salt
 	dyna_salt_init(self);
-	if (!self->params.tests[0].fields[1])
-		self->params.tests[0].fields[1] = self->params.tests[0].ciphertext;
-	ciphertext = self->methods.prepare(self->params.tests[0].fields, self);
-	ciphertext = self->methods.split(ciphertext, 0, self);
-	salt = self->methods.salt(ciphertext);
-	if (salt)
-		dyna_salt_create(salt);
+	if (self->methods.tunable_cost_value[0] && autotune_db->real) {
+		struct db_main *db = autotune_db->real;
```

Comment:
The real db availability in the first reset() call (via *real) solves all problems. I was not aware of it.
- I will run all 'my' tests against it. But, I see no problems in autotune code. 

Maybe one concern: are you sure there is no "fast" format with cost and 100 millions of loaded hashes?